### ISSUE TITLE
Better support for union values.

### ISSUE THREAD

**Issue by philandstuff** [philandstuff/dhall-golang#13]
Currently, there is no way for a user to unpack a value of union type from a Value.

This also means there's no way to unmarshal a union value into a go data structure.

**Comment by terinjokes** [comment]
Yep, just ran into this. `unionVal` isn't exported, so there's nothing I can do if `Eval` returns it.
